contracts/utils/SignatureValidator.sol

```diff
@@ -99,4 +100,41 @@ contract SignatureValidator {
 
     return signer;
   }
+
+ /**
+   * @notice Returns true if the provided signature is valid for the given signer.
+   * @dev Supports SignatureType.EIP712, SignatureType.EthSign, and ERC1271 signatures
+   * @param _hash      Hash that was signed
+   * @param _signer    Address of the signer candidate
+   * @param _signature Signature byte array
+   */
+  function isValidSignature(
+    bytes32 _hash,
+    address _signer,
+    bytes memory _signature
+  ) internal view returns (bool valid) {
+    uint256 signatureType = uint8(_signature[_signature.length - 1]);
```

Comment:
Where do we say that the signature type is the last byte? We also use the term "Signature type" for two difference purposes: EIP712/EthSign/ERC1271 and for FLAG_SIGNATURE/FLAG_ADDRESS/FLAG_DYNAMIC_SIGNATURE. Should use different terms and make sure both are added to comments on how signatures are constructed.

Reply:
I think we exclusively use `item type` for `FLAG_SIGNATURE/FLAG_ADDRESS/FLAG_DYNAMIC_SIGNATURE`, because it can also be used to denote plain addresses, I agree we need to be very specific on the documentation.